packages/cli/src/controllers/nodeDetails.controller.ts

```diff
@@ -0,0 +1,113 @@
+import { Service } from 'typedi';
+import { NextFunction, Response } from 'express';
+import type {
+	INodeListSearchResult,
+	INodePropertyOptions,
+	ResourceMapperFields,
+} from 'n8n-workflow';
+import { jsonParse } from 'n8n-workflow';
+import { LoadNodeDetails } from 'n8n-core';
+
+import { Authorized, Get, Middleware, RestController } from '@/decorators';
+import {
+	NodeListSearchRequest,
+	NodeParameterOptionsRequest,
+	NodesOptionsRequest,
+	ResourceMapperRequest,
+} from '@/requests';
+import { getBase } from '@/WorkflowExecuteAdditionalData';
+import { NodeTypes } from '@/NodeTypes';
+import { BadRequestError } from '@/ResponseHelper';
+
+@Service()
+@Authorized()
+@RestController('/node-details')
```

Comment:
"Details" is rather imprecise. Suggestions:

- `/node-details` → `/dynamic-node-parameters`
- `/node-details/parameter-options` → `/dynamic-node-parameters/options`
- `/node-details/list-search` → `/dynamic-node-parameters/resource-locator-results`
- `/node-details/mapping-fields` → `/dynamic-node-parameters/resource-mapper-fields`

Or: `/remote-node-parameters`

Reply:
how about just `/node-parameters` ?

Reply:
So my reasoning for the long name is that "node parameters" are all of them (most being static and already in nodes-base) whereas this controller deals with the subset of node parameters that are dynamically generated, fetched from a remote resource.